gateways/checkoutcom/request_builder.go

```diff
@@ -0,0 +1,61 @@
+package checkoutcom
+
+import (
+	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	checkout_com_common "github.com/checkout/checkout-sdk-go/common"
+	"github.com/checkout/checkout-sdk-go/payments"
+)
+
+func buildChargeParams(authRequest *sleet.AuthorizationRequest) (*payments.Request, error) {
+	var source = payments.CardSource{
+		Type: "card",
+		Number: authRequest.CreditCard.Number,
+		ExpiryMonth: uint64(authRequest.CreditCard.ExpirationMonth),
+		ExpiryYear: uint64(authRequest.CreditCard.ExpirationYear),
+		Name: authRequest.CreditCard.FirstName + " " + authRequest.CreditCard.LastName,
+		CVV: authRequest.CreditCard.CVV,
+		BillingAddress: &checkout_com_common.Address{
+			AddressLine1: common.SafeStr(authRequest.BillingAddress.StreetAddress1),
+			AddressLine2: common.SafeStr(authRequest.BillingAddress.StreetAddress2),
+			City:         common.SafeStr(authRequest.BillingAddress.Locality),
+			State:        common.SafeStr(authRequest.BillingAddress.RegionCode),
+			ZIP:          common.SafeStr(authRequest.BillingAddress.PostalCode),
+			Country:      common.SafeStr(authRequest.BillingAddress.CountryCode),
+		},
+	}
+
+	return &payments.Request{
+		Source:   source,
+		Amount:   uint64(authRequest.Amount.Amount),
+		Capture:  common.BPtr(false),
+		Currency: authRequest.Amount.Currency,
+		Reference: *authRequest.ClientTransactionReference,
+		Customer: &payments.Customer{
+			Email: common.SafeStr(authRequest.BillingAddress.Email),
+			Name:  authRequest.CreditCard.FirstName + " " + authRequest.CreditCard.LastName,
+		},
+	}, nil
+}
+
+func buildRefundParams(refundRequest *sleet.RefundRequest) (*payments.RefundsRequest, error) {
+	return &payments.RefundsRequest{
+		Amount:    uint64(refundRequest.Amount.Amount),
+		Reference: *refundRequest.ClientTransactionReference,
```

Comment:
Do we want to use `TransactionReference` here? (Same for Capture)

Reply:
So transaction reference in this case is a pay id which is included in the url of the request. That's what I'm saving in transaction reference. The reference field they provide is supposed to be for our internal tracking reference, so I used client id here. The checkout.com folks pointed this out when I was passing pay id in the url and the reference field.

Reply:
Swapped to merchant order ref based on later comments